src/converter/renderers/render-aem.js

```diff
@@ -14,20 +15,124 @@ import {
   decodeCQMetadata,
   generateHash,
 } from './utils/aem-page-meta-utils.js';
+import { getMatchLanguageForTag } from '../../common/utils/language-utils.js';
 import { getMetadata, setMetadata } from '../modules/utils/dom-utils.js';
-import { writeStringToFileAndGetPresignedURL } from '../../common/utils/file-utils.js';
+import {
+  writeStringToFileAndGetPresignedURL,
+  readFile,
+} from '../../common/utils/file-utils.js';
+import stateLib from '../../common/utils/state-lib-util.js';
 
 export const aioLogger = Logger('render-aem');
 
 const byteSize = (str) => new Blob([str]).size;
 const isLessThanOneMB = (str) => byteSize(str) < 1024 * 1024 - 1024; // -1024 for good measure :)
 
+const mapTagsToTitles = (meta, taxonomyData) => {
+  let locTitles = [];
+  if (!meta?.length) return [];
+  if (!Array.isArray(taxonomyData) || !taxonomyData?.length) {
+    aioLogger.error('Invalid or empty taxonomy data:', taxonomyData);
+    return [];
+  }
+
+  const tags = meta.split(',').map((tag) => tag.trim());
+  locTitles = tags
+    .map((tag) => {
+      const match = taxonomyData.find((item) => item.tag.trim() === tag);
+      return match ? match.title : null;
+    })
+    .filter(Boolean);
+
+  if (locTitles.length) {
+    locTitles = locTitles.join(', ');
+  }
+  return locTitles;
+};
+
+// Fetch tags data from AEM taxonomy sheets
+async function fetchTaxonomyData(pagePath, params, taxonomy) {
+  const state = await stateLib.init();
```

Comment:
FYI statelib max value is 1mb.

https://developer.adobe.com/app-builder/docs/guides/application_state/#limits--validation

unless you plan on storing every key, you will run into this limit, might be better to use Lib Files for this:

https://github.com/adobe/aio-lib-files

See files-util.js